### PR TITLE
Add tracer utils

### PR DESCRIPTION
**Pull request by dafortune**

### CODE REVIEW

File: lib/tracer.js [hunk 1]
```diff
@@ -19,6 +20,7 @@ const Tags = Object.assign({
   AUTH0_CHANNEL: constants.TAG_AUTH0_CHANNEL,
   AUTH0_HOSTNAME: constants.TAG_AUTH0_HOSTNAME,
   AUTH0_PURPOSE: constants.TAG_AUTH0_PURPOSE,
+  AUTH0_REQUEST_ID: constants.TAG_AUTH0_REQUEST_ID,
```
Comment (dm03514) [inline lib/tracer.js]: Is the request id to correlate traces and logs? Are services currently using/passing request ids?

Comment (dafortune) [reply lib/tracer.js]: exactly

File: lib/tracer_utils.js [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+}
+
+exports.mapToTags = function mapToTags(map) {
+  let keys = Object.keys(map);
+
+  if (keys.length > MAX_OBJECT_KEYS) {
+    keys = keys.splice(0, MAX_OBJECT_KEYS);
+  }
+
+  return keys.reduce((tags, key) => {
```
Comment (jaredsuttles) [inline lib/tracer_utils.js]: Minor comment: personally don't like using `reduce` here as IMO it's misleading - could just as well use `forEach`

File: lib/tracer_utils.js [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+  } else if (typeof data === 'object') {
+    return serializeObject(data);
+  } else if (typeof data === 'number') {
+    return serializeNumber(data);
+  } else if (typeof data === 'string') {
+    return serializeString(data);
+  }
+}
+
+exports.mapToTags = function mapToTags(map) {
```
Comment (dm03514) [inline lib/tracer_utils.js]: <3

File: lib/tracer_utils.js (hunk 2 above)
Comment (dafortune) [reply lib/tracer_utils.js]: You could think you are reducing a potentially deep map into a single level map. Having said that I don't personally care about this kind of subtle things, let me change it by a forEach

**Pull request merged**